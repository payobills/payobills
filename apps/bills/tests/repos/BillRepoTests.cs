namespace payobills.bills.tests.repos;

using payobills.bills.dtos;
using payobills.bills.models;
using payobills.bills.repos;
using payobills.bills.svc;
using Snapshooter.Xunit;

public class BillRepoTests
{
    private readonly BillRepo billRepo;

    public BillRepoTests()
    {
        this.billRepo = new BillRepo(
            new HelperGuidService(),
            new HelperDateTimeService()
        );
    }

    [Fact]
    public async Task ShouldCreateBill_SnapshotTest()
    {
        Bill addedBill;
        var billDto = new BillDto
        {
            Name = "Electricity - Home",
            BillingDate = 3,
            PayByDate = 17,
        };

        addedBill = await billRepo.AddBillAsync(billDto);

        Snapshot.Match(addedBill);
    }
}