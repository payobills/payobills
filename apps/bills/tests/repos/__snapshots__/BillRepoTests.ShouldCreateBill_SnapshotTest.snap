{
  "Id": "00000000-0000-0000-0000-000000000000",
  "Name": "Electricity - Home",
  "BillingDate": 3,
  "PayByDate": 17,
  "LatePayByDate": 0,
  "CreatedAt": "2022-01-01T00:00:00",
  "UpdatedAt": "2022-01-01T00:00:00"
}
